carapace-server/src/main/java/org/carapaceproxy/server/RequestHandler.java

```diff
@@ -386,6 +392,42 @@ private void serveStaticMessage() {
         }
     }
 
+    private void serveRedirect() {
+        DefaultFullHttpResponse res = new DefaultFullHttpResponse(
+                HttpVersion.HTTP_1_1,
+                HttpResponseStatus.valueOf(action.errorcode < 0 ? 302 : action.errorcode) // redirect: 3XX
+        );
+        addCustomResponseHeaders(res);
+        res.headers().set(HttpHeaderNames.CACHE_CONTROL, "no-cache");
+
+        String location = action.redirectLocation;
+        String host = this.connectionToClient.getListenerHost();
```

Comment:
do not use this info

Reply:
I would start with the original Host used by the client, 
we can use this information in case the "Host" header is not available